feat(@embark/cli): repl support inside dashboard

PascalPrecht opened the pull request:
This is a rebased version of @lastmjs 's effort in #804 

@lastmjs FYI: unfortunately the only way to update your PR was to create a new one as we don't have write access to your fork.

@andremedeiros @iurimatias can we put together a list of outstanding things in this effort? Is this here still up to date? https://github.com/embark-framework/embark/pull/804#issuecomment-420424512

Closes #768

StatusSceptre commented:
Looks like @lastmjs put a bit of effort into this one in a few separate issues @PascalPrecht do you agree that we should issue the bounty for the time /  effort put in so far? @lastmjs thanks for all the effort I know it's been quite a mammoth of a task.

PascalPrecht commented:
@lastmjs I fully understand. Unfortunately we didn't seem to be as responsive on this one as we should've been either. Sorry it's been like that.

Obviously, we'd be happy to get your further support to finish this feature, but we also understand that you have other priorities.

Thank you so much for your efforts so far!

@StatusSceptre I agree this shouldn't have been for nothing and also, it's on us that this one hasn't progressed as quickly as desired.

PascalPrecht commented on src/cmd/dashboard/dashboard.js:
```diff
@@ -60,7 +60,8 @@ class Dashboard {
 
-    this.events.setCommandHandler("console:command", monitor.executeCmd.bind(monitor));
+    //TODO Should we setup a handler for logger:log or something similar?
+    //TODO So everywhere that we call logText.this.logger, we can just raise an event?
```
I think this needs clarity. Not sure what's meant by that.

PascalPrecht commented on src/cmd/dashboard/monitor.js:
```diff
@@ -36,7 +35,64 @@ class Monitor {
+        // else {
+        //   logText.log(groups[1]);
+        //   terminal.write(groups[2]);
+        // }
```
Here's a lot of commented code. This needs to go out.

PascalPrecht commented on src/cmd/dashboard/monitor.js:
```diff
@@ -36,7 +35,64 @@ class Monitor {
+
+    // setTimeout(() => {
+    //   console.error('monkey')
+    // }, 5000);
```
More commented code here

PascalPrecht commented on src/cmd/dashboard/monitor.js:
```diff
@@ -287,97 +343,41 @@ class Monitor {
+        },
+        scrollable: false,
+        handler: (data) => {
+          this.terminalReadableStream.push(data);
```
Some of those layout changes affect the "Available Services" box, which now bleeds into the Logs section

PascalPrecht commented on src/cmd/dashboard/monitor.js:
```diff
@@ -261,11 +282,9 @@ class Monitor {
+      padding: 1,
       width: "100%",
-      height: "60%",
+      height: "70%",
```
Weird. 🤷‍♂️

PascalPrecht commented:
@jrainville I remember to tried this PR and something still wasn't as expected. Did you have time to write down the feedback somewhere?

jrainville commented:
Yes. I was about to say, I didn't see the REPL when testing, but just had a flash. I didn't start babel when testing.

So I just tested again and I do have the REPL now. However, it gets stuck because of the `Error: Cannot find module 'C:\dev\embark\modules\debugger'` issue. You might need to rebase.

I commented the debbugger just to be able to test and the REPL works. There is just a overlap issue at the top:
![image](https://user-images.githubusercontent.com/11926403/49378452-9f41e780-f6da-11e8-810a-08219e9b4654.png)

PascalPrecht commented:
Thanks for the update @jrainville I rebased it now on top of latest `master`.

So the problem with the layouting is that I ran into the exact same issue and adjusted the numbers to make it work. However, those numbers don't seem to fit on bigger resolutions.

As mentioned in: https://github.com/embark-framework/embark/pull/1078#pullrequestreview-178757145 it's kind of unpredictable because neo-blessed does weird things with math.

Do you think you can take a look at this as you have a bigger screen at hand?

PascalPrecht commented:
I tested this locally again. The boxes in the top right corner don't fit 100% but it's not a big deal at all. I guess we can clean up the commits and merge this.

Thank you for the fine-tuning @jrainville !
